sympy/solvers/diophantine/diophantine.py

```diff
@@ -51,6 +51,97 @@
     "univariate"}
 
 
+class DiophantineSolutionSet(set):
+    """
+    Container for a set of solutions to a particular diophantine equation.
+
+    The base representation is a set of tuples representing each of the solutions.
+
+    Parameters
+    ==========
+
+    symbols : list
+        List of free symbols in the original equation.
+    parameters: list (optional)
+        List of parameters to be used in the solution
+
+    Examples
+    ========
+
+    Adding solutions:
+
+        >>> from sympy.solvers.diophantine.diophantine import DiophantineSolutionSet
+        >>> from sympy.abc import x, y, t, u
+        >>> s1 = DiophantineSolutionSet([x, y], [t, u])
+        >>> s1
+        set()
+        >>> s1.add_solution(2, 3)
+        >>> s1.add_solution(-1)
```

Comment:
this makes me nervous...what if you wanted to add a value for `u`? or wouldn't you want to do that?

Reply:
You're right, I think there's no need to accept incorrect length solutions.